paymentsheet/src/main/java/com/stripe/android/paymentsheet/elements/common/DropdownConfigInterface.kt

```diff
@@ -0,0 +1,15 @@
+package com.stripe.android.paymentsheet.elements.common
+
+internal interface DropdownConfigInterface {
+    val debugLabel: String
+
+    /** This is the label to describe the element */
+    val label: Int
+
+    fun convertToDisplay(paramFormatted: String?): String = paramFormatted ?: ""
+    fun convertToPaymentMethodParam(displayFormatted: String): String? {
```

Comment:
> I noticed this when I was looking at `IdealBankConfig` as it overwrote this `convertToPaymentMethodParam`. Are all `DropdownConfigInterface` implementations going to relate to payment methods? We might want to rework this interface if its purpose is to be generic for use with all types of drop down widgets and not only payment methods.

I only have plans to use this for payment methods.  There are several payment methods that will need bank selectors: FPX?, EPS?, Ideal, P24?, and all of these will be used by the PaymentMethod Params.

We don't really have drop downs anywhere else.  I would say if we have another use case for the drop down that we abstract it at that point, if needed.

Reply:
> Have you considered using the index as the argument for both `convertToDisplay(...)` and `convertToPaymentMethodParam(...)`? I think we might be adding extra constraints here that limit how we can work with this interface. Both implementations (`IdealBankConfig` and `CountryConfig`) end up having 2 maps to hold the display -> value and value -> display relationships. We could make things a bit more direct by passing the index and having the implementation figure out the display or value to return (quite simple if we have something backed by a list):
> 
> ```
> fun convertToDisplay(index: Int?): String
> fun convertToPaymentMethodParam(index: Int): String?
> ```
> 
> On the topic of using a list, the `convertToPaymentMethod(...)` might even be replaced with `getItem(index: Int): T?` if you want to make this interface generic (separate the data and presentation models). My understanding is that this method is used to provide the value we want to use for the business logic (as opposed to presentation. logic). We could use `data class IdealBankData(val value: String, val label: String)` or the `Country` data class you already have for the implementations of this interface.
> 
> This might be a very confusing comment as I ended up conflating 2 different points here: "use the index instead of the display value" and "we can make this interface generic while we are at it". Let me know if you want to chat about this.

I used an index here as it might be the case that the display needs to show an icon plus text, or something else, and by using an index rather than a string they could get the full class.